src/Raven.Server/Documents/Queries/QueryBuilderHelper.cs

```diff
@@ -132,6 +133,11 @@ public static long GetLongValue(Query query, QueryMetadata metadata, BlittableJs
         }
     }
 
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    internal static (object Value, ValueTokenType Type)
+        GetValue(QueryEnvironment queryEnvironment, QueryExpression expression, bool allowObjectsInParameters = false) => GetValue(
```

Comment:
Passing around this `QueryEnvironment` if there is a failed attempt at inlining will be deadly.